src/java.net.http/share/classes/jdk/internal/net/http/PullPublisher.java

```diff
@@ -25,62 +25,71 @@
 
 package jdk.internal.net.http;
 
-import java.util.Iterator;
 import java.util.concurrent.Flow;
+
 import jdk.internal.net.http.common.Demand;
 import jdk.internal.net.http.common.SequentialScheduler;
 
 /**
- * A Publisher that publishes items obtained from the given Iterable. Each new
- * subscription gets a new Iterator.
+ * A {@linkplain Flow.Publisher publisher} that publishes items obtained from the given {@link CheckedIterable}.
+ * Each new subscription gets a new {@link CheckedIterator}.
  */
 class PullPublisher<T> implements Flow.Publisher<T> {
 
-    // Only one of `iterable` and `throwable` can be non-null. throwable is
+    // Only one of `iterable` or `throwable` should be null, and the other non-null. throwable is
     // non-null when an error has been encountered, by the creator of
     // PullPublisher, while subscribing the subscriber, but before subscribe has
     // completed.
-    private final Iterable<T> iterable;
+    private final CheckedIterable<T> iterable;
     private final Throwable throwable;
 
-    PullPublisher(Iterable<T> iterable, Throwable throwable) {
+    PullPublisher(CheckedIterable<T> iterable, Throwable throwable) {
```

Comment:
Looking at the usages of this constructor, would it be better if we got rid of this 2-arg constructor and introduced a new
```java
PullPublisher(Throwable throwable) {
}
```

That way all we have to do is verify that the incoming argument is non-null, both in this constructor and the other existing `PullPublisher(CheckedIterable<T> iterable)`